File: lat2_c++/hardware.cpp
#include <iostream>
#include <string>

using namespace std;

class hardware : public product{
private:
	string brand;
	string model;

public:
	hardware(){
		this->brand = "";
		this->model = "";
	}

	hardware(string brand, string model){
		this->brand = brand;
		this->model = model;
	}

	void setbrand(string brand){
		this->brand = brand;
	}

	void setmodel(string model){
		this->model = model;
	}


	string getbrand(){
		return brand;
	}

	string getmodel(){
		return model;
	}

	void printHardware(){
		cout << "Brand : ";
		cout << getbrand() << endl;
		cout << "Model : ";
		cout << getmodel() << endl;
	}

	~hardware(){}
};

File: lat2_PHP/hardware.php
<?php

class hardware extends product{

	private $brand = "";
	private $model = "";

	public function __construct($brand = "", $model = ""){
		$this->brand = $brand;
		$this->model = $model;
	}

	public function setbrand($brand){
		$this->brand = $brand;
	}

	public function getbrand(){
		return $this->brand;
	}

	public function setmodel($model){
		$this->model = $model;
	}

	public function getmodel(){
		return $this->model;
	}

	public function printHardware(){
		echo "Brand 		: ". getbrand() ."<br>";
		echo "Model 		: ". getmodel() ."<br>";
	}
}